### PR TITLE
ShiftCyclicPath

### PR DESCRIPTION
**Pull request by MeikeWeiss**
Issue #101

### CODE REVIEW

**Review by TomGoertzen — CHANGES_REQUESTED** [review]
Since ShiftCyclicPath start with two tests (in paths.gi) you should also implement an NC version of this method

File: gap/Paths/paths.gd [hunk 1]
```diff
@@ -434,6 +434,32 @@ DeclareAttribute( "EdgesAsPerm", IsVertexEdgePath );
...
+#! >    [[1,2,5],[2,3,6],[3,4,7],[1,4,8]] );;
+#! gap> path:=VertexEdgePathByEdges(fourGon,[5,6,7,8]);
+#! ( v2, E5, v3, E6, v4, E7, v5, E8, v2 )
+#! gap> ShiftCyclicPath(path,4,6);
+#! ( v4, E6, v3, E5, v2, E8, v5, E7, v4 )
+#! @EndExampleSession
+#!
+#! @Returns a vertex-edge-path
+#! @Arguments path, vertex, edge
+DeclareOperation("ShiftCyclicPath", [IsVertexEdgePath, IsPosInt, IsPosInt]);
```
Comment (markusbaumeister) [inline gap/Paths/paths.gd:460]: NC version here

File: gap/Paths/paths.gi [hunk 2]
```diff
@@ -334,6 +334,56 @@ InstallMethod(ConcatenationOfPaths, "for a twisted polygonal complex and two ver
...
+			
+	    if not IsDuplicateFree(path) then
+		Error(Concatenation("ShiftCyclicPath: Given vertex-edge-path ",String(path)," is not duplicate free."));
+	    fi;
+		
+            complex:=AssociatedPolygonalComplex(path);
+	    path:=PathAsList(path);
+			
+	    if path[1] = vertex and path[2] = edge then
+                return VertexEdgePath(complex,path);
```
Comment (markusbaumeister) [inline gap/Paths/paths.gi:355]: It would be nice if there was a test case for this situation.

File: gap/Paths/paths.gi [hunk 3]
```diff
@@ -552,6 +602,56 @@ InstallMethod(ConcatenationOfPaths, "for a twisted polygonal complex and two edg
...
+                if path[i] = face then
+                    fPos := i;
+                fi;
+            od;
+            Assert(0, fPos > 0);
+
+            if ePos + 1 = fPos then
+                return EdgeFacePath(complex,Concatenation( path{[ePos..Length(path)-1]}, path{[1..ePos]} ));
+            elif fPos + 1 = ePos then
+                return EdgeFacePath(complex, Reversed( Concatenation( path{[ePos..Length(path)]}, path{[2..ePos]} ) ));
```
Comment (markusbaumeister) [inline gap/Paths/paths.gi:649]: This case is also not covered.

**Review by markusbaumeister — CHANGES_REQUESTED** [review]
This looks good so far. Could you please add some tests for the remaining corner cases?

**Review by TomGoertzen — APPROVED** [review]
Looks good. Thank you for the changes :)

**Review by markusbaumeister — APPROVED** [review]
Very nice :).
